autonat.go

```diff
@@ -336,6 +362,15 @@ func (as *AmbientAutoNAT) probeNextPeer() {
 		if proto, err := as.host.Peerstore().SupportsProtocols(p, AutoNATProto); len(proto) == 0 || err != nil {
 			continue
 		}
+		goodAddr := false
+		for _, a := range info.Addrs {
+			if !skipDial(as.host, a) {
+				goodAddr = true
+			}
+		}
+		if !goodAddr {
+			continue
```

Comment:
This will dial them if there exists at least one "good" address. The check is going to have to be a bit more complicated:

1. Skip dialing if there are no public/non-relay addresses (currently works).
2. Skip dialing if the remote peer advertises an IP address that (a) is public and (b) matches one of ours (doesn't currently work).